README.md

```diff
@@ -3,741 +3,470 @@
+-   `CompactRegistered(address indexed sponsor, bytes32 claimHash, bytes32 typehash)`: Emitted when a compact is registered via `register`, `registerMultiple`, or combined deposit-and-register functions. ([`ITheCompact.sol#L63`](./src/interfaces/ITheCompact.sol#L63))
+-   `AllocatorRegistered(uint96 allocatorId, address allocator)`: Emitted when a new allocator is registered via [`__registerAllocator`](./src/interfaces/ITheCompact.sol#L561). ([`ITheCompact.sol#L70`](./src/interfaces/ITheCompact.sol#L70))
+-   `EmissaryAssigned(address indexed sponsor, bytes12 indexed lockTag, address emissary)`: Emitted when a sponsor assigns or changes an emissary via [`assignEmissary`](./src/interfaces/ITheCompact.sol#L532). ([`EmissaryLib.sol#L46`](./src/lib/EmissaryLib.sol#L46))
+
+Standard `ERC6909.Transfer` events are also emitted for mints, burns, and transfers of resource lock tokens.
+
+## Key Data Structures
+Many functions in The Compact use custom structs for their calldata. Here are some of the most important ones:
+
+-   **For Claims (passed to `ITheCompactClaims` functions):**
+    -   `Claim` ([`src/types/Claims.sol`](./src/types/Claims.sol)): For claims involving a single resource lock on a single chain.
+        ```solidity
+        // Defined in src/types/Claims.sol
+        struct Claim {
+            bytes allocatorData;
+            bytes sponsorSignature;
+            address sponsor;
+            uint256 nonce;
+            uint256 expires;
+            bytes32 witness;
+            string witnessTypestring;
+            uint256 id;
+            uint256 allocatedAmount;
+            Component[] claimants;
+        }
+        ```
+    -   `BatchClaim` ([`src/types/BatchClaims.sol`](./src/types/BatchClaims.sol)): For multiple resource locks on a single chain.
+    -   `MultichainClaim` ([`src/types/MultichainClaims.sol`](./src/types/MultichainClaims.sol)): For single resource lock claims on the notarized (i.e., origin) chain of a multichain compact.
+    -   `ExogenousMultichainClaim` ([`src/types/MultichainClaims.sol`](./src/types/MultichainClaims.sol)): For single resource lock claims on an exogenous chain (i.e., any chain _other than_ the notarized chain).
+    -   `BatchMultichainClaim` ([`src/types/BatchMultichainClaims.sol`](./src/types/BatchMultichainClaims.sol)): For multiple resource locks on the notarized chain.
+    -   `ExogenousBatchMultichainClaim` ([`src/types/BatchMultichainClaims.sol`](./src/types/BatchMultichainClaims.sol)): For multiple resource locks on an exogenous chain.
+    -   `BatchClaimComponent` ([`src/types/Components.sol`](./src/types/Components.sol)): Used within batch claim structs.
+        ```solidity
+        // Defined in src/types/Components.sol
+        struct BatchClaimComponent {
+            uint256 id;
+            uint256 allocatedAmount;
+            Component[] portions;
```

Comment:
We should probably show Component prior to this

Reply:
It is presented earlier on line 206